apmproxy/apmserver.go

```diff
@@ -60,11 +62,17 @@ func (c *Client) ForwardApmData(ctx context.Context) error {
 			return nil
 		case data := <-c.AgentDataChannel:
 			if err := c.forwardAgentData(ctx, data); err != nil {
+				if errors.Is(err, accumulator.ErrNoData) {
+					c.logger.Debugf("Received something from '%s' without APMData", data.AgentInfo)
+					continue
+				}
 				return err
 			}
-			// Wait for metadata to be available, metadata will be available as soon as
-			// the first agent data is processed.
-			lambdaDataChan = c.LambdaDataChannel
+			once.Do(func() {
+				// Wait for metadata to be available, metadata will be available as soon as
+				// the first agent data is processed.
+				lambdaDataChan = c.LambdaDataChannel
+			})
 		case data := <-lambdaDataChan:
```

Comment:
If `c.LambdaDataChannel` doesn't change then why don't we use it directly? Unless there are hidden semantics, both the channel assignment and the `Once.Do` are essentially NOPs. The comment is confusing me, as there's no waiting we're just assigning a reference.

```suggestion
		case data := <-c.lambdaDataChan:
```